nyxx_interactions/lib/src/Interactions.dart

```diff
@@ -65,63 +72,100 @@ class Interactions {
     });
   }
 
+  Future<void> _syncPermissions() async {
+    final commandPartition = _partition<SlashCommandBuilder>(
+        this._commandBuilders, (element) => element.guild == null);
+    final globalCommands = commandPartition.first;
+    final groupedGuildCommands =
+        _groupSlashCommandBuilders(commandPartition.last);
+
+    await this
+        ._client
+        .httpEndpoints
+        .sendRawRequest("/applications/${this._client.app.id}/commands/permissions", "PUT",
+            body: [globalCommands.where((builder) => builder.permissions.isNotEmpty).map((builder) => {
+              "id": builder._id, 
+              "permissions": [for (final permsBuilder in builder.permissions) permsBuilder.build()]
+            })]);
+
+    for (final entry in groupedGuildCommands.entries) {
+      await this._client.httpEndpoints.sendRawRequest("/applications/${this._client.app.id}/guilds/${entry.key}/commands/permissions", "PUT",
+            body: [entry.value.where((builder) => builder.permissions.isNotEmpty).map((builder) => {
+              "id": builder._id, 
+              "permissions": [for (final permsBuilder in builder.permissions) permsBuilder.build()]
+            })]);
+    }
+  }
+
   /// Syncs command builders with discord immediately.
   /// Warning: Client could not be ready at the function execution.
   /// Use [syncOnReady] for proper behavior
   Future<void> sync() async {
-    final commandPartition = _partition<SlashCommandBuilder>(this._commandBuilders, (element) => element.guild == null);
+    final commandPartition = _partition<SlashCommandBuilder>(
+        this._commandBuilders, (element) => element.guild == null);
     final globalCommands = commandPartition.first;
-    final groupedGuildCommands = _groupSlashCommandBuilders(commandPartition.last);
+    final groupedGuildCommands =
+        _groupSlashCommandBuilders(commandPartition.last);
 
-    final globalCommandsResponse = await this._client.httpEndpoints.sendRawRequest(
-        "/applications/${this._client.app.id}/commands",
-        "PUT",
-        body: [
-          for(final builder in globalCommands)
-            builder.build()
-        ]
-    );
+    final globalCommandsResponse = await this
+        ._client
+        .httpEndpoints
+        .sendRawRequest("/applications/${this._client.app.id}/commands", "PUT",
+            body: [for (final builder in globalCommands) builder.build()]);
 
     if (globalCommandsResponse is HttpResponseSuccess) {
+      final body = globalCommandsResponse.jsonBody as List<dynamic>;
+      var i = 0;
+      for (final commandData in body) {
+        globalCommands.elementAt(i)._setId(Snowflake(commandData["id"]));
+        i++;
+      }
```

Comment:
why it is done that way? Are you relying that results returned from api are in order? If so that could led to erros because api is "eventually consistent" and could happen that results are out of order

Reply:
Fixed in latest commit.